pkg/alertmanager/operator.go

```diff
@@ -426,11 +428,14 @@ func (c *Operator) sync(key string) error {
 
 	ssetClient := c.kclient.AppsV1().StatefulSets(am.Namespace)
 	// Ensure we have a StatefulSet running Alertmanager deployed.
-	obj, exists, err = c.ssetInf.GetIndexer().GetByKey(alertmanagerKeyToStatefulSetKey(key))
-	if err != nil {
+	obj, err := c.ssetInfs.Get(alertmanagerKeyToStatefulSetKey(key))
+
+	if err != nil && !apierrors.IsNotFound(err) {
 		return errors.Wrap(err, "retrieving statefulset failed")
 	}
 
+	exists := !apierrors.IsNotFound(err)
+
 	if !exists {
```

Comment:
here we are negating a negation. It would be much clearer if we simply wrote:
```go
if apierrors.IsNotFound(err) {
}
```

Reply:
agreed, and i think i did this only due to cherry-pick friendliness, but the inline version should be as friendly.

Reply:
fixed